### Analyse data Titanic Disaster

### Model: SVM classification

In [1]:
# setup absolute path to location of package Starts and config-file 
from inspect import getsourcefile
import os.path as path, sys
current_dir = path.dirname(path.abspath(getsourcefile(lambda:0)))
sys.path.insert(0, current_dir[:current_dir.rfind(path.sep)])

# import package Starts
from Starts.startmodskl import *

local_kwargs {'data_path': './data/train.csv, ./data/titanic_no_nan.csv ,./data/test.csv', 'drop_obj_col': False, 'nan_drop_col': False, 'nan_drop_row': False, 'nan_zero': False, 'nan_mean': True, 'nan_mean_neighbors': False}


In [2]:
nonan_data = idata[1]
nonan_data.head()

Survived  Pclass  Sex        Age  Ticket      Fare  Title  FamilySize
0         0       3    1  29.699118     199    7.2292      5           0
1         1       3    0   0.750000     194   19.2583      4           3
2         0       3    1  29.699118     487    6.8583      5           0
3         1       1    1  36.000000      33  120.0000      5           3
4         0       2    1  29.699118     629   15.0500      5           0

In [3]:
# Choose method Support Vector Machine method to classify data
classifier, y_true, y_pred = StartModSKL.classification_svm(nonan_data, dependent_label='Survived', kernel='rbf')

classifier

SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,
  decision_function_shape='ovr', degree=3, gamma='auto', kernel='rbf',
  max_iter=-1, probability=False, random_state=100, shrinking=True,
  tol=0.001, verbose=False)

In [4]:
# Compare the predicted value with true_value
len(y_true), len(y_pred)

(179, 179)

In [5]:
confusion_matrix(y_true, y_pred)

array([[108,   4],
       [ 60,   7]])

### Result: 
#### the correct predicted result 107, 
#### the wrong predicted result 72

In [6]:
# Choose method Support Vector Machine method to classify data
classifier, y_true, y_pred = StartModSKL.classification_svm(nonan_data, 'Survived', kernel='linear')

classifier

SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,
  decision_function_shape='ovr', degree=3, gamma='auto', kernel='linear',
  max_iter=-1, probability=False, random_state=100, shrinking=True,
  tol=0.001, verbose=False)

In [7]:
confusion_matrix(y_true, y_pred)

array([[95, 17],
       [23, 44]])

### Result: 
#### the correct predicted result 139, 
#### the wrong predicted result 40

In [8]:
# Choose method Support Vector Machine method (Kernel='sigmoid') to classify data
classifier, y_true, y_pred = StartModSKL.classification_svm(nonan_data, dependent_label='Survived', 
                                                            kernel='sigmoid')

classifier

SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,
  decision_function_shape='ovr', degree=3, gamma='auto', kernel='sigmoid',
  max_iter=-1, probability=False, random_state=100, shrinking=True,
  tol=0.001, verbose=False)

In [9]:
len(y_true), len(y_pred)

(179, 179)

In [10]:
confusion_matrix(y_true, y_pred)

array([[112,   0],
       [ 67,   0]])

### Result: 
#### the correct predicted result 112
#### the wrong predicted result 67